### PR TITLE
API worker for non-blocking HTTP requests.

### PR DESCRIPTION
**Pull request by iivanou**

### CODE REVIEW

**Comment by iivanou** [comment]
> I wonder if we better use event-based control?
> 
> * I see we already do, why do we need sentinel in this case?

No need. Agree.

File: reportportal_client/core/worker.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            logger.debug('[%s] Current status for tasks is: {%s} unfinished',
+                         self.name, self._data_queue.unfinished_tasks)
+
+        if cmd == ControlCommand.STOP:
+            request = self._request_get()
+            while request is not None:
+                self._request_process(request)
+                request = self._request_get()
+
+        if cmd and cmd.is_stop_cmd():
```
Comment (Scanters) [inline reportportal_client/core/worker.py:84]: `if cmd` case is covered in line 70 and may be removed
In general, how about calling `self._stop()` at the end of line 78: `if cmd == ControlCommand.STOP:`?

File: reportportal_client/core/worker.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        """
+        if not cmd:
+            return  # No command received
+
+        logger.debug('[%s] Processing {%s} command', self.name, cmd)
+        if cmd == ControlCommand.REPORT_STATUS:
+            logger.debug('[%s] Current status for tasks is: {%s} unfinished',
+                         self.name, self._data_queue.unfinished_tasks)
+
+        if cmd == ControlCommand.STOP:
```
Comment (Scanters) [inline reportportal_client/core/worker.py:78]: I believe we can use `if cmd.is_stop_cmd():` here

File: reportportal_client/core/worker.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    def send_command(self, cmd):
+        """Send control command to the worker queue."""
+        self._cmd_queue.put(cmd)
+
+    def send_request(self, request):
+        """Send a request to the worker queue.
+
+        :param request: RPRequest object
+        """
+        self._data_queue.put((request.priority, request))
```
Comment (Scanters) [inline reportportal_client/core/worker.py:145]: We won't need to separate priority if we implement a priority field comparator for the request (__lt__)
```
from queue import PriorityQueue


class Request:
    def __init__(self, weight=1):
        self.weight = weight

    def __lt__(self, other):
        return self.weight < other.weight

    def __str__(self):
        return f"Request<{self.weight}>"


q = PriorityQueue()

q.put(Request(2))
q.put(Request(3))
q.put(Request(4))
q.put(Request(1))
q.put(Request(4))
q.put(Request(5))

while True:
    print(q.get())
```

>>>
```
Request<1>
Request<2>
Request<3>
Request<4>
Request<4>
Request<5>
```

**Review by Scanters — CHANGES_REQUESTED** [review]

**Review by Scanters — APPROVED** [review]

**Pull request merged**